### ISSUE TITLE
Layout/EmptyLineBetweenMethodDefs does not require a line between methods separated by another method call

### ISSUE THREAD

**Issue by jviney** [rubocop/rubocop#10058]
Layout/EmptyLineBetweenMethodDefs permits the following code:

```ruby
def foo
end
call_something("here")
def bar
end
```

## Describe the solution you'd like

The rule should be triggered in the example above.

**Comment by koic** [comment]
On the other hand, method defs above methods like `alias` and `alias_method` can be allowed by default.

```ruby
# good
def foo
end
alias bar foo

def baz
end
```

**Comment by hirasawayuki** [comment]
Is it correct that the rule is triggered as follows?

```ruby
# bad
def foo
end
call_something("here")
def bar
end

# bad
def foo
end
call_something("here")

def bar
end

# bad
def foo
end

call_something("here")
def bar
end

# good
def foo
end

call_something("here")

def bar
end
```

**Comment by caalberts** [comment]
I see this requirement as similar to `EmptyLinesAroundAccessModifier`.

```rb
# bad
def foo
end
private
def bar
end

# good
# bad
def foo
end

private

def bar
end
```

 Maybe this should be implemented as a separate cop targeting method calls between method definitions instead. Perhaps a `Layout/EmptyLineBetweenMethodCallAndDefinition`
